lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -1349,7 +1349,7 @@ void VariableDeclarator(Schema type, List<Field> fields):
       if ("order".equals(key))
         order = Field.Order.valueOf(getTextProp(key,props,token).toUpperCase());
 
-    boolean validate = !SchemaResolver.isUnresolvedSchema(type);
+    boolean validate = SchemaResolver.isFullyResolvedSchema(type);
```

Comment:
This ensures we only validate fields if the schema is (recursively) resolved.

Note that although this triggers validation less often, every default value is still validated because `ResolvingVisitor` copies the entire schema, and thus (re)validates all default values.